iree/samples/static_library/static_library_demo.c

```diff
@@ -0,0 +1,204 @@
+  }
+
+  // Populate initial values for 4 * 2 = 8.
+  const int kElementCount = 4;
+  iree_hal_dim_t shape[1] = {kElementCount};
+  iree_hal_buffer_view_t* arg0_buffer_view = NULL;
+  iree_hal_buffer_view_t* arg1_buffer_view = NULL;
+  float kFloat4[] = {4.0f, 4.0f, 4.0f, 4.0f};
+  float kFloat2[] = {2.0f, 2.0f, 2.0f, 2.0f};
+
+  iree_hal_memory_type_t input_memory_type =
+      IREE_HAL_MEMORY_TYPE_HOST_LOCAL | IREE_HAL_MEMORY_TYPE_DEVICE_VISIBLE;
+  if (iree_status_is_ok(status)) {
+    iree_hal_buffer_view_clone_heap_buffer(
+        iree_hal_device_allocator(device), shape, IREE_ARRAYSIZE(shape),
+        IREE_HAL_ELEMENT_TYPE_FLOAT_32, input_memory_type,
+        IREE_HAL_BUFFER_USAGE_ALL,
+        iree_make_const_byte_span((void*)kFloat4,
+                                  sizeof(float) * kElementCount),
+        &arg0_buffer_view);
+  }
+  if (iree_status_is_ok(status)) {
+    iree_hal_buffer_view_clone_heap_buffer(
+        iree_hal_device_allocator(device), shape, IREE_ARRAYSIZE(shape),
+        IREE_HAL_ELEMENT_TYPE_FLOAT_32, input_memory_type,
+        IREE_HAL_BUFFER_USAGE_ALL,
+        iree_make_const_byte_span((void*)kFloat2,
+                                  sizeof(float) * kElementCount),
+        &arg1_buffer_view);
+  }
+
+  // Queue buffer views for input.
+  if (iree_status_is_ok(status)) {
+    status =
+        iree_runtime_call_inputs_push_back_buffer_view(&call, arg0_buffer_view);
+  }
+  iree_hal_buffer_view_release(arg0_buffer_view);
+
+  if (iree_status_is_ok(status)) {
+    status =
+        iree_runtime_call_inputs_push_back_buffer_view(&call, arg1_buffer_view);
+  }
+  iree_hal_buffer_view_release(arg1_buffer_view);
+
+  // Invoke call.
+  if (iree_status_is_ok(status)) {
+    status = iree_runtime_call_invoke(&call, /*flags=*/0);
+  }
+
+  // Retreive output buffer view with results from the invocation.
+  iree_hal_buffer_view_t* ret_buffer_view = NULL;
+  if (iree_status_is_ok(status)) {
+    status = iree_runtime_call_outputs_pop_front_buffer_view(&call,
+                                                             &ret_buffer_view);
+  }
+
+  // Read back the results and ensure we got the right values.
+  iree_hal_buffer_mapping_t mapped_memory;
+  if (iree_status_is_ok(status)) {
+    status = iree_hal_buffer_map_range(
+        iree_hal_buffer_view_buffer(ret_buffer_view),
+        IREE_HAL_MEMORY_ACCESS_READ, 0, IREE_WHOLE_BUFFER, &mapped_memory);
+  }
+  for (int i = 0; i < mapped_memory.contents.data_length / sizeof(float); ++i) {
+    if (((const float*)mapped_memory.contents.data)[i] != 8.0f) {
+      return status;
+    }
+  }
+
+  // Cleanup call and buffers.
+  iree_hal_buffer_unmap_range(&mapped_memory);
+  iree_hal_buffer_view_release(ret_buffer_view);
+  iree_runtime_call_deinitialize(&call);
+
+  // Cleanup session and instance.
+  iree_runtime_session_release(session);
+  iree_runtime_instance_release(instance);
+
+  return iree_ok_status();
```

Comment:
Return the status you've been updating throughout. Otherwise, the fallthrough case will appear successful.

```suggestion
  return status;
```